test/bun.js/install/bun-link.test.ts

```diff
@@ -0,0 +1,201 @@
+import { file, spawn } from "bun";
+import {
+  afterEach,
+  beforeEach,
+  expect,
+  it,
+} from "bun:test";
+import { bunExe } from "bunExe";
+import { bunEnv as env } from "bunEnv";
+import { access, mkdir, mkdtemp, readdir, readlink, rm, writeFile } from "fs/promises";
+import { basename, join } from "path";
+import { tmpdir } from "os";
+
+let package_dir, link_dir;
+
+beforeEach(async () => {
+  link_dir = await mkdtemp(join(tmpdir(), "bun-link.test"));
+  package_dir = await mkdtemp(join(tmpdir(), "bun-link.pkg"));
+});
+afterEach(async () => {
+  await rm(link_dir, { force: true, recursive: true });
+  await rm(package_dir, { force: true, recursive: true });
+});
+
+it("should link package", async () => {
+  var link_name = basename(link_dir).slice("bun-link.".length);
+  await writeFile(join(link_dir, "package.json"), JSON.stringify({
+    name: link_name,
+    version: "0.0.1",
+  }));
+  await writeFile(join(package_dir, "package.json"), JSON.stringify({
+    name: "foo",
+    version: "0.0.2",
+  }));
+
+  const { stdout: stdout1, stderr: stderr1, exited: exited1 } = spawn({
+    cmd: [bunExe(), "link"],
+    cwd: link_dir,
+    stdout: null,
+    stdin: "pipe",
+    stderr: "pipe",
+    env,
+  });
+  expect(stderr1).toBeDefined();
+  const err1 = await new Response(stderr1).text();
+  expect(err1.replace(/^(.*?) v[^\n]+/, "$1").split(/\r?\n/)).toEqual([
+    "bun link",
+    "",
+  ]);
+  expect(stdout1).toBeDefined();
+  expect(await new Response(stdout1).text()).toContain(`Success! Registered "${link_name}"`);
+  expect(await exited1).toBe(0);
+
+  const { stdout: stdout2, stderr: stderr2, exited: exited2 } = spawn({
+    cmd: [bunExe(), "link", link_name],
+    cwd: package_dir,
+    stdout: null,
+    stdin: "pipe",
+    stderr: "pipe",
+    env,
+  });
+  expect(stderr2).toBeDefined();
+  const err2 = await new Response(stderr2).text();
+  expect(err2.replace(/^(.*?) v[^\n]+/, "$1").split(/\r?\n/)).toEqual([
+    "bun link",
+    "",
+  ]);
+  expect(stdout2).toBeDefined();
+  const out2 = await new Response(stdout2).text();
+  expect(out2.replace(/\s*\[[0-9\.]+ms\]\s*$/, "").split(/\r?\n/)).toEqual([
+    "",
+    ` installed ${link_name}@link:${link_name}`,
+    "",
+    "",
+    " 1 packages installed",
```

Comment:
@Jarred-Sumner not the most important of issues, but I failed to understand why `bun link` seems to insert extra `\n` somewhere compared to `bun install`:

https://github.com/oven-sh/bun/blob/2edbf4d0ec7340d40f5e51b3b7d262499b5bc131/test/bun.js/install/bun-install.test.ts#L231-L235